app/src/main/java/org/oppia/app/player/state/StateAdapter.kt

```diff
@@ -0,0 +1,86 @@
+package org.oppia.app.player.state
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.RecyclerView
+import androidx.databinding.library.baseAdapters.BR
+import kotlinx.android.synthetic.main.state_button_item.view.*
+import org.oppia.app.R
+import org.oppia.app.player.state.itemviewmodel.StateButtonViewModel
+import org.oppia.app.player.state.listener.ButtonInteractionListener
+import org.oppia.app.databinding.StateButtonItemBinding
+
+const val VIEW_TYPE_CONTENT = 1
+const val VIEW_TYPE_INTERACTION_READ_ONLY = 2
+const val VIEW_TYPE_NUMERIC_INPUT_INTERACTION = 3
+const val VIEW_TYPE_TEXT_INPUT_INTERACTION = 4
+const val VIEW_TYPE_STATE_BUTTON = 5
+
+class StateAdapter(
+  private val itemList: MutableList<Any>,
+  private val buttonInteractionListener: ButtonInteractionListener
```

Comment:
Note for @BenHenning : This `ButtonInteractionListener` was earlier introduced in #244 which was merged directly after review from @veena14cs just to move ahead with code.

Reply:
Ack, thanks for the heads up!